tox.ini

```diff
@@ -99,24 +99,20 @@ commands =
 
 [testenv:codechecks]
 basepython = python3
+skip_install = true
 deps =
-     # Make sure we get a version of PyLint that respects config, and isn't too
-     # old.
-     pylint < 2.18, >2.14
+     ruff
```

Comment:
The version in setup.py is pinned precisely but this one is unpinned.

It's not even entirely obvious to me why we have this (or why we had pyflakes) as a dependency in setup.py.  So devs could get the tools with `pip install ...[test]` I guess?  It would be nice to declare this only once and (I guess?) for the declaration to include a version pin.

Reply:
I have a follow-up PR that I'll open once this one is merged which adds pinning.

Reply:
Actually, no, I didn't. So I'll make them match.

Reply:
The reason for duplication is faster CI; it allows not installing everything on each CI run, just ruff. Not quite sure how to fix that... maybe a `.ruff.version` file?

Reply:
I guess setup.py could extract the ruff version from tox.ini with a regex but oh god.